Source/PluginProcessor.h

```diff
@@ -111,28 +130,54 @@ class PinkTromboneADSR
 				envelopeVal = 1.0f;
 				goToNextState();
 			}
+			
+			if (parameters.attackExp >= 0) scaledEnvelopeVal = powf(envelopeVal, parameters.attackExp);
+			else scaledEnvelopeVal = powf(envelopeVal, (-1/parameters.attackExp));
 		}
+		
 		else if (state == State::decay)
 		{
 			envelopeVal -= decayRate;
+			
+			if (parameters.decayExp >= 0) scaledEnvelopeVal = powf(envelopeVal, parameters.decayExp);
+			else {
+				if (envelopeVal >= 0) scaledEnvelopeVal = powf(envelopeVal, (-1/parameters.decayExp));
+				else scaledEnvelopeVal = 0;
+			}
+			
+			scaledEnvelopeVal = parameters.sustain + (1 - parameters.sustain)*scaledEnvelopeVal;
 
-			if (envelopeVal <= parameters.sustain)
+			if (scaledEnvelopeVal <= parameters.sustain)
 			{
 				envelopeVal = parameters.sustain;
+				scaledEnvelopeVal = parameters.sustain;
 				goToNextState();
 			}
-			if (parameters.decayExp >= 0)
-				scaledEnvelopeVal = powf(envelopeVal, parameters.decayExp);
-			else
-				scaledEnvelopeVal = powf(envelopeVal, (-1/parameters.decayExp));
-			return scaledEnvelopeVal;
 		}
+		
 		else if (state == State::sustain)
 		{
 			envelopeVal = parameters.sustain;
+			scaledEnvelopeVal = parameters.sustain;
+		}
+		
+		else if (state == State::release)
+		{
+			envelopeVal -= releaseRate;
+			
+			if (parameters.releaseExp >= 0) scaledEnvelopeVal = powf(envelopeVal, parameters.releaseExp);

```

Comment:
I think this is wrong, and that once you enter sustain, you actually want to set envelopeVal to 1. Suppose sustain is 0.5 and the releaseExp is 2. When you start the release, the envelope should be at 0.5 (the sustain). However, it looks like it will instead be at (0.5 ^ 2), which is 0.25. I think you want to set envelopeVal to 1 in the sustain phase, and multiply `powf(envelopeVal, parameters.releaseExp)` by `parameters.sustain` in the release phase.